### PR TITLE
Add ViT support

### PR DESCRIPTION
**Pull request by tomtseng**
Associated KataGo-custom PR: https://github.com/AlignmentResearch/KataGo-custom/pull/100

### CODE REVIEW

File: configs/compute/1gpu.cfg [hunk 1]
```diff
@@ -1,11 +1,11 @@
-numGameThreads = 128
+numGameThreads = 256
```
Comment (AdamGleave) [inline configs/compute/1gpu.cfg:1]: Is this better for all models or just for ViT models? If latter, do we need to make it conditional on model type? If former, perhaps we should split this into a separate PR?

File: configs/compute/1gpu.cfg [hunk 2]
```diff
@@ -21,5 +21,8 @@ nnRandomize = true
 deviceToUseThread0 = 0
 
 useFP16 = true
-# We only set NHWC for CUDA because TensorRT requires useNHWC == false.
-cudaUseNHWC = true
+# The TorchScript backend requires NCHW.
```
Comment (AdamGleave) [inline configs/compute/1gpu.cfg:24]: Is this harmless for non-ViT models (b/c we're not using TensorRT backend) or could it have some performance implications?

File: kubernetes/shuffle-and-export.sh [hunk 3]
```diff
@@ -1,14 +1,39 @@
...
-cd /engines/KataGo-tensorflow/python
+
+USE_TORCHSCRIPT=0
+# Command line flag parsing (https://stackoverflow.com/a/33826763/4865149).
+# Flags must be specified before positional arguments.
+while [ -n "${1-}" ]; do
+  case $1 in
+    # Use PyTorch for training rather than TensorFlow.
+    --use-pytorch) USE_PYTORCH=1 ;;
+    # Export models to C++ as TorchScript models instead than using KataGo's
```
Comment (AdamGleave) [inline kubernetes/shuffle-and-export.sh:10]:
```suggestion
    # Export models to C++ as TorchScript models instead of using KataGo's
```

File: kubernetes/shuffle-and-export.sh [hunk 4]
```diff
@@ -1,14 +1,39 @@
...
 DIRECTORY="$2"
 VOLUME_NAME="$3"
 # 1 to enable gatekeeper, 0 to disable gatekeeper
 USE_GATING="$4"
 
+if [ -n "${USE_PYTORCH:-}" ]; then
+  cd /engines/KataGo-custom/python
+else
+  if [ "${USE_TORCHSCRIPT:-}" -eq 1 ]; then
+    echo "Error: --use-pytorch is required if --use-torchscript is set."
```
Comment (AdamGleave) [inline kubernetes/shuffle-and-export.sh:29]: Missing something to cause script to abort?
```suggestion
    echo "Error: --use-pytorch is required if --use-torchscript is set."
    exit 1
```

**Review by AdamGleave — COMMENTED** [review]
Initial review, looks mostly good, a few nits.

Guess we'll want to finish PR in KataGo-custom before merging this -- though if waiting for a while may want to try and get the KataGo-custom PR over the finish line sooner and split off enhancements into separate PRs.

File: configs/compute/1gpu.cfg (hunk 2 above)
Comment (tomtseng) [reply configs/compute/1gpu.cfg]: yeah, I should've benchmarked this. Seems like this makes CUDA-backend models ~30% slower, so I'll revert this and leave a comment that this param should be toggled for TorchScript-backend models. Might be good for me to just override NHWC to be false in the C++ code for TorchScript models, but I notice the TensorRT backend also just throws an error if NHWC is set incorrectly so I don't feel too bad about not implementing forceful overriding

**Review by ed1d1a8d — APPROVED** [review]
LGTM
